astropy/utils/diff.py

```diff
@@ -99,27 +91,30 @@ def report_diff_values(a, b, fileobj=sys.stdout, indent_width=0):
                 indent_width))
         return num_diffs == 0
 
-    padding = max(len(typea.__name__), len(typeb.__name__)) + 3
+    typea = type(a)
+    typeb = type(b)
```

Comment:
Here `a` and `b` have been converted to string, so this is not the same implementation as before, maybe the cause of the failure?